eng/Versions.props

```diff
@@ -123,16 +123,16 @@
     <MicrosoftWin32RegistryVersion>5.0.0</MicrosoftWin32RegistryVersion>
     <StyleCopAnalyzersVersion>1.2.0-beta.507</StyleCopAnalyzersVersion>
     <SystemBuffersVersion>4.5.1</SystemBuffersVersion>
-    <SystemCollectionsImmutableVersion>7.0.0</SystemCollectionsImmutableVersion>
+    <SystemCollectionsImmutableVersion>8.0.0</SystemCollectionsImmutableVersion>
     <SystemComponentModelAnnotationsVersion>5.0.0</SystemComponentModelAnnotationsVersion>
     <SystemDataSqlClientVersion>4.8.6</SystemDataSqlClientVersion>
     <SystemDrawingCommonVersion>8.0.0</SystemDrawingCommonVersion>
     <SystemIOFileSystemAccessControlVersion>5.0.0</SystemIOFileSystemAccessControlVersion>
     <SystemMemoryVersion>4.5.5</SystemMemoryVersion>
     <SystemReflectionMetadataVersion>9.0.0-alpha.1.24072.1</SystemReflectionMetadataVersion>
     <!-- Keep toolset versions in sync with dotnet/msbuild and dotnet/sdk -->
-    <SystemReflectionMetadataToolsetVersion>7.0.0</SystemReflectionMetadataToolsetVersion>
-    <SystemReflectionMetadataLoadContextVersion>7.0.0</SystemReflectionMetadataLoadContextVersion>
+    <SystemReflectionMetadataToolsetVersion>8.0.0</SystemReflectionMetadataToolsetVersion>
```

Comment:
Based on the comment above, this toolset version should not be updated without coordination with msbuild and sdk repos.

@ViktorHofer, FYI, this is blocking the codeflow https://github.com/dotnet/sdk/pull/38665 (while previous change made it to the next phases without problems https://github.com/dotnet/sdk/pull/38553). We can either revert this or update https://github.com/dotnet/sdk/blob/b32df9afff77e8f1afbd417e7c70841867cb6cdc/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj#L75 to 8.0 (if we choose the latter, we can close https://github.com/dotnet/runtime/issues/96809).

Reply:
I coordinated with people from the msbuild team at multiple points while developing this along with the SBRP team, so I'm sorry that something fell through the cracks here. Is there a specific step we missed?

Reply:
I see now from looking through the replies to Larry's comment on this issue that you told us there. For some reason, GitHub did not send me a notification for your comment, so I didn't see it :( I guess in the future blocking issues like that need to be PR comments instead of replies.

Reply:
My preference would definitely be to take the bump in the sdk. VS 17.9 is the where these changes will ship and it has taken the update.

cc @ViktorHofer @marcpopMSFT

Reply:
NP, https://github.com/dotnet/runtime/issues/96809 is tracking it, but it seems (almost) resolved by your PR. Once confirmed, we can just update [sdk's Microsoft.NET.Build.Tasks.csproj#L74-L76](https://github.com/dotnet/sdk/blob/b32df9afff77e8f1afbd417e7c70841867cb6cdc/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj#L74-L76) to 8.0 (now that all the related toolings are using 8.0; msbuild, HostModel and VS).

Reply:
I **think** for this to work, sdk would need to upgrade its .NET Framework leg to a VS 17.9 Preview image. I'm not sure if such is already available. Worst case, we would need to revert this dependency bump here. @lewing can you please make sure that this dependency flow issue gets resolved: https://github.com/dotnet/sdk/pull/38665?

Reply:
Correct me if I'm wrong but the sdk _build_ should be fine, it'd just need the testing legs to use VS17.9 Preview right? That should already be the case: https://github.com/dotnet/sdk/blob/0962c1f89f5daf924a9fe876c80e80b0bde63b0d/.vsts-ci.yml#L108-L111

I'll try bumping the .csproj in the sdk PR and see what happens.

Reply:
Correct. Unification at run time.

Reply:
Looks like it worked :)